test/monster_test/monster_test.c

```diff
@@ -1270,13 +1344,39 @@ int test_scan_by(flatcc_builder_t *B)
         goto done;
     }
     if (3 != (pos = nsc(uint8_vec_scan_ex(inv, 3, nsc(end), 3)))) {
-        printf("scan_from(item) not working for repeating item of inventory\n");
+        printf("scan_ex(item) not working for repeating item of inventory\n");
         goto done;
     }
     if (5 != (pos = nsc(uint8_vec_scan_ex(inv, 4, nsc(end), 3)))) {
-        printf("scan_from(item+1) not working for repeating item of inventory\n");
+        printf("scan_ex(item+1) not working for repeating item of inventory\n");
+        goto done;
+    }
+    if (5 != (pos = nsc(uint8_vec_rscan(inv, 3)))) {
+        printf("rscan not working for repeating item of inventory\n");
         goto done;
     }
+    if (3 != (pos = nsc(uint8_vec_rscan_ex(inv, 1, 4, 3)))) {
+        printf("rscan_ex not working for repeating item of inventory\n");
+        goto done;
+    }
+
+    /* Test that all scan functions are generated for string arrays */
+    nsc(string_vec_scan(strings, "Hello"));
+    nsc(string_vec_scan_ex(strings, 0, nsc(end), "Hello"));
+    nsc(string_vec_scan_n(strings, "Hello", 4));
+    nsc(string_vec_scan_ex_n(strings, 0, nsc(end), "Hello", 4));
+    nsc(string_vec_rscan(strings, "Hello"));
+    nsc(string_vec_rscan_ex(strings, 0, nsc(end), "Hello"));
+    nsc(string_vec_rscan_n(strings, "Hello", 4));
+    nsc(string_vec_rscan_ex_n(strings, 0, nsc(end), "Hello", 4));
+
+#ifdef FLATCC_ALLOW_SCAN_FOR_ALL_FIELDS
```

Comment:
This should not be #ifdef

The configuration assigns 0 or 1 so it can be overridden by compile flag.
#if FLATCC_ALLOW_SCAN_FOR_ALL_FIELDS
...
#endif